### ISSUE TITLE
Unhandled 'error' event: Error: Batch is already running

### ISSUE THREAD

**Issue by CodeBrauer** [dynasticdevelop/place#63]
Just played around with 2 users:
```
POST /api/place 200 7.950 ms - 54
POST /api/place 200 5.879 ms - 54
POST /api/place 200 10.919 ms - 54
POST /api/place 200 10.752 ms - 54
events.js:163
      throw er; // Unhandled 'error' event
      ^

Error: Batch is already running
    at /Users/gw/dev/place/node_modules/pajk-lwip/lib/BatchPrototypeInit.js:53:39
    at Object.exec (/Users/gw/dev/place/node_modules/decree/dist/cjs/decree.js:69:29)
    at Batch.exec (/Users/gw/dev/place/node_modules/pajk-lwip/lib/BatchPrototypeInit.js:52:16)
    at user.addPixel (/Users/gw/dev/place/util/PaintingHandler.js:109:57)
    at /Users/gw/dev/place/models/user.js:156:20
    at /Users/gw/dev/place/node_modules/mongoose/lib/model.js:3720:16
    at /Users/gw/dev/place/node_modules/mongoose/lib/services/model/applyHooks.js:146:20
    at _combinedTickCallback (internal/process/next_tick.js:73:7)
    at process._tickCallback (internal/process/next_tick.js:104:9)
```

Using node 7.8.0 on macOS 10.10

**Comment by nullpixel** [comment]
Heh, and we thought this was fixed. I'll look into it.

**Comment by nullpixel** [comment]
This bug is far from easy to fix, actually, and it has some pretty big implications on when it occurs: it slows down the server side operations whilst the error happens.

**Comment by AppleBetas** [comment]
I've been thinking about how I want to implement the fix for this bug, and believe I have decided. Using a smart queuing method that automatically pushes requests for batch images to the end if they aren't in progress (on a new pixel) and makes them wait if one is, I believe that we can significantly decrease the server load created when painting a pixel (which is already pretty small as it is, but we need to be able to scale properly), and wipe out this issue.

**Comment by nullpixel** [comment]
Yeah, that sounds good to me

**Comment by EricRabil** [comment]
A project tracker for this issue has been opened [here](https://github.com/dynasticdevelop/place/projects/2)

**Comment by AppleBetas** [comment]
Merged with #122
